clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -39,6 +40,26 @@ public Struct(Schema schema) {
         this.values = new Object[this.schema.numFields()];
     }
 
+    /**
+     * Returns the serialized byte buffer
+     */
+    public ByteBuffer getSerialized() {
+        if (serialized == null) {
+            return null;
+        }
+        return serialized.duplicate();
+    }
+
+    /**
+     * Sets the serialized byte buffer
+     */
+    public void setSerialized(ByteBuffer serialized) {
+        if (serialized == null) {
+            throw new NullPointerException("The input serialized byte buffer cannot be null");
+        }
+        this.serialized = serialized;
```

Comment:
will this buffer ever be explicitly returned/destroyed, or are we waiting for GC ?

Reply:
We depend on GC to reclaim it.